packages/list-plugins/README.md

```diff
@@ -103,3 +103,39 @@ const { createdBy, updatedBy } = require('@keystonejs/list-plugins');
 ```
 
 _Note_: The API is the same.
+
+# singleton
+
+This plugin makes a list singleton by allowing only one item in the list. Useful for list which must contain only one items.
+
+## Usage
+
+```js
+const { singleton } = require('@keystonejs/list-plugins');
+
+keystone.createList('ListWithPlugin', {
+  fields: {...},
+  plugins: [
+    singleton({...}),
+  ],
+});
+```
+
+## Config
+
+| Option          | Type      | Default | Description                                       |
+| --------------- | --------- | ------- | ------------------------------------------------- |
+| `preventDelete` | `Boolean` | `true`  | Prevents deletion of the (only) item in the list. |
```

Comment:
I think you are right, we can keep this restricted to create only when there is no item. 
one caveat is that if `create` is set to false it is rejected before we hit any hooks.

one improvement would be that we pre-create this item, which I don't see a way without having an event system like `keystone.on('connect', () => createItem` from within plugins.
(this event was there in v4, if I remember it right)

Reply:
Could we do create-object-if-missing operation during the `update` mutation, rather than having it be a system startup thing (which is a pattern I'd rather avoid in general).

Reply:
how does that work? update mutation needs an ID and in admin-ui you can not update an item which is not there

Reply:
create on read is another possibility? there is no read hook for list. will have to add infrastructure for this to create an item if list is empty.

Reply:
Hmm, I see that you're right about update needing an `ID`. I think you're right, we probably do need to support the `create` operation, anything other than that will require too much jumping through hoops.

Could you update the PR so that it's got `{ delete: false }` enforced and then we can see what if there's anything else that needs tweaking.

Reply:
working...

Reply:
see #3011  and #3012
I will split remaining work for singleton in new PR.